packages/lib/src/index.ts

```diff
@@ -34,7 +13,42 @@ export type HTMLElementProps<
   ? Props
   : never;
 
-export { typeIs, testHelpers };
+/**
+ * Helper that constructs a type requiring at least one of the passed keys
+ * to be present in the passed interface.
+ *
+ * Example
+ * ```
+ * interface ExampleInterface {
+ *   alwaysRequired: boolean,
+ *   sometimesRequired: boolean,
+ *   requiredOtherTimes: boolean,
+ * }
+ *
+ * type ExampleEither = Either<ExampleInterface, 'sometimesRequired' | 'requiredOtherTimes'>
+ *
+ * // The above is equivalent to:
+ * interface SharedInExampleInterface {
+ *   alwaysRequired: boolean,
+ * }
+ *
+ * interface FirstIsRequired extends SharedInExampleInterface {
+ *   sometimesRequired: boolean,
+ *   requiredOtherTimes?: boolean,
+ * }
+ *
+ * interface SecondIsRequired extends SharedInExampleInterface {
+ *   sometimesRequired?: boolean,
+ *   requiredOtherTimes: boolean,
+ * }
+ *
+ * type EquivalentToExampleEither = FirstIsRequired | SecondIsRequired
+ * ```
+ */
+export type Either<T, Keys extends keyof T = keyof T> = Omit<T, Keys> &
+  {
+    [K in Keys]-?: Required<Pick<T, K>> & Partial<Pick<T, Exclude<Keys, K>>>;
```

Comment:
So just so I understand... the `-?:` here is removing the optionality from `keyof T` that's being "looped" through here, not from any of the keys that are actually output, right? Essentially just avoiding having an additional `| undefined` in the generated type.

Reply:
That's my understanding. In all fairness, this is a modified version of an answer in this SO thread: https://stackoverflow.com/questions/40510611/typescript-interface-require-one-of-two-properties-to-exist

I didn't know what `-?` did in TS until I found this hah